src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -50,36 +50,71 @@ class FeatureFlagValue:
         Represents if the Feature flag is On/Off/Conditionally On
     :ivar dict {string, FeatureFilter[]} conditions:
         Dictionary that contains client_filters List (and server_filters List in future)
-    '''
-
-    def __init__(self,
-                 id_,
-                 description=None,
-                 enabled=None,
-                 conditions=None,
-                 display_name=None):
+    :ivar FeatureAllocation allocation:
+        Determines how variants should be allocated for the feature to various users.
+    :ivar list FeatureVariant[] variants:
+        The list of variants defined for this feature.
+    :ivar dict telemetry:
+        The declaration of options used to configure telemetry for this feature.
+    """
+
+    def __init__(
+        self,
+        id_,
+        description=None,
+        enabled=None,
+        conditions=None,
+        display_name=None,
+        variants=None,
+        allocation=None,
+        telemetry=None,
+    ):
         default_conditions = {FeatureFlagConstants.CLIENT_FILTERS: []}
 
         self.id = id_
         self.description = "" if description is None else description
         self.enabled = enabled if enabled else False
         self.conditions = conditions if conditions else default_conditions
-        self.display_name = display_name
+        if allocation is not None:
```

Comment:
Is there a reason why we do not just set these properties to `None`?

Reply:
My thinking was not to add new properties to a flag that doesn't have the new properties. If we set them to None the feature flag value would have these properties set to null in its value e.g `{id : "test", description" : "this is a description", ... allocation: null, telemetry: null }`

Reply:
In my opinion, we should be clear about what properties a class has, else we might end up with `AttributeError` when using the object in other parts of the code without adding a `hasattr` check. 

If anything at all, we can write custom serialization logic that will be passed to the `default` kwarg of `json.dumps` to avoid having `none` for certain properties. Thoughts?